lib/mnesia/src/mnesia.erl

```diff
@@ -2361,96 +2531,180 @@ load_mnesia_or_abort() ->
 
+-spec backup(Dest::term(), Mod::module()) ->
+                    'ok' | {'error', Reason::term()}.
 backup(Opaque, Mod) ->
     mnesia_log:backup(Opaque, Mod).
 
+-spec traverse_backup(Src::term(), Dest::term(), Fun, Acc) ->
+                             {'ok', Acc} | {'error', Reason::term()} when
+      Fun :: fun((Items, Acc) -> {Items,Acc}).
 traverse_backup(S, T, Fun, Acc) ->
     mnesia_bup:traverse_backup(S, T, Fun, Acc).
 
+-spec traverse_backup(Src::term(), SrcMod::module(),
+                      Dest::term(), DestMod::module(),
+                      Fun, Acc) ->
+                             {'ok', Acc} | {'error', Reason::term()} when
+      Fun :: fun((Items, Acc) -> {Items,Acc}).
 traverse_backup(S, SM, T, TM, F, A) ->
     mnesia_bup:traverse_backup(S, SM, T, TM, F, A).
 
+-spec install_fallback(Src::term()) -> 'ok' | {'error', Reason::term()}.
 install_fallback(Opaque) ->
     mnesia_bup:install_fallback(Opaque).
 
+-spec install_fallback(Src::term(), Mod::module()|[Opt]) ->
+                              'ok' | {'error', Reason::term()} when
+      Opt :: Module | Scope | Dir,
+      Module :: {'module', Mod::module()},
+      Scope :: {'scope', 'global' | 'local'},
+      Dir :: {'mnesia_dir', Dir::string()}.
 install_fallback(Opaque, Mod) ->
     mnesia_bup:install_fallback(Opaque, Mod).
 
+-spec uninstall_fallback() -> 'ok' | {'error', Reason::term()}.
 uninstall_fallback() ->
     mnesia_bup:uninstall_fallback().
 
+-spec uninstall_fallback(Args) -> 'ok' | {'error', Reason::term()} when
+      Args :: [{'mnesia_dir', Dir::string()}].
 uninstall_fallback(Args) ->
     mnesia_bup:uninstall_fallback(Args).
 
+-spec activate_checkpoint([Arg]) -> {'ok', Name, [node()]} | {'error', Reason::term()} when
+      Arg :: {'name', Name} | {'max', [table()]} | {'min', [table()]} |
+             {'allow_remote', boolean()} | {'ram_overrides_dump', boolean()}.
 activate_checkpoint(Args) ->
     mnesia_checkpoint:activate(Args).
 
+-spec deactivate_checkpoint(Name::_) -> 'ok' | {'error', Reason::term()}.
 deactivate_checkpoint(Name) ->
     mnesia_checkpoint:deactivate(Name).
 
+-spec backup_checkpoint(Name::_, Dest::_) -> 'ok' | {'error', Reason::term()}.
 backup_checkpoint(Name, Opaque) ->
     mnesia_log:backup_checkpoint(Name, Opaque).
 
+-spec backup_checkpoint(Name::_, Dest::_, Mod::module()) ->
+                               'ok' | {'error', Reason::term()}.
 backup_checkpoint(Name, Opaque, Mod) ->
     mnesia_log:backup_checkpoint(Name, Opaque, Mod).
 
+-spec restore(Src::_, [Arg]) -> t_result([table()]) when
+      Op  :: 'skip_tables' | 'clear_tables' | 'keep_tables' | 'restore_tables',
+      Arg :: {'module', module()} | {Op, [table()]} | {'default_op', Op}.
 restore(Opaque, Args) ->
     mnesia_schema:restore(Opaque, Args).
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% Table mgt
 
+-spec create_table([Arg]) -> t_result('ok') when
+      Arg :: {'name', table()} |
```

Comment:
There is a lot of redundancy here, for essentially the same functions.  I suggest you introduce a type for the common parts of the Arguments.  Also, perhaps the `'set' | 'ordered_set' | 'bag'` part can be a remote type from `ets` (`duplicate_bag` is not allowed here?).